### PR TITLE
Separate the chain-index indexers into multiple individual parts

### PR DESCRIPTION
**Pull request by koslambrou**
* Chain index memory and sqlite indexers are separated

* Refactored indexer modules for chain-index

Pre-submit checklist:
- Branch
    - [ ] Tests are provided (if possible)
    - [ ] Commit sequence broadly makes sense
    - [ ] Key commits have useful messages
    - [ ] Relevant tickets are mentioned in commit messages
    - [ ] Formatting, materialized Nix files, PNG optimization, etc. are updated
- PR
    - [ ] Self-reviewed the diff
    - [ ] Useful pull request description
    - [ ] Reviewer requested

### CODE REVIEW

File: plutus-chain-index-core/src/Plutus/ChainIndex.hs [hunk 1]
```diff
@@ -68,8 +71,16 @@ handleChainIndexEffects RunRequirements{trace, stateMVar, conn, securityParam} a
...
+
+        -- All indexers
+        $ interpret (handleUtxoIndexer chainIndexDb)
+        $ interpose (handleDatumFromHashIndexer chainIndexDb)
+        $ interpose (handleRedeemerFromHashIndexer chainIndexDb)
+        $ interpose (handleScriptFromHashIndexer chainIndexDb)
+        $ interpose (handleUtxoFromAddressIndexer chainIndexDb)
+        $ interpose (handleAssetClassFromAddressIndexer chainIndexDb)
+
+        $ interpret (handleQuery chainIndexDb)
```
Comment (ak3n) [inline plutus-chain-index-core/src/Plutus/ChainIndex.hs:83]: Would be nice to have a wrapper around `chainIndexDb` but not necessary.

File: plutus-chain-index-core/src/Plutus/ChainIndex/Indexer/Sqlite/Handlers.hs [hunk 2]
```diff
@@ -73,90 +94,83 @@ handleQuery ::
...
+    )
+    => db (DatabaseEntity Sqlite db)
+    -> ChainIndexQueryEffect ~> Eff effs
+handleQuery db = \case
+    DatumFromHash dh            -> getDatumFromHash db dh
+    ValidatorFromHash hash      -> getScriptFromHash db hash
+    MintingPolicyFromHash hash  -> getScriptFromHash db hash
+    RedeemerFromHash hash       -> getRedeemerFromHash db hash
+    StakeValidatorFromHash hash -> getScriptFromHash db hash
+    UnspentTxOutFromRef tor     -> getUtxoutFromRef db tor
```
Comment (ak3n) [inline plutus-chain-index-core/src/Plutus/ChainIndex/Indexer/Sqlite/Handlers.hs:107]: Maybe move `DatabaseSettings` to a reader monad?

File: plutus-chain-index-core/src/Plutus/ChainIndex.hs [hunk 3]
```diff
@@ -54,24 +58,53 @@ runChainIndexEffects runReq action =
...
+    => Eff (ChainIndexControlEffect ': effs) ~> Eff effs
+handleChainIndexControl =
+    interpret
+        (\eff ->
+            handleUtxoIndexer chainIndexDb eff
+         >> handleDatumFromHashIndexer chainIndexDb eff
+         >> handleRedeemerFromHashIndexer chainIndexDb eff
+         >> handleScriptFromHashIndexer chainIndexDb eff
+         >> handleUtxoFromAddressIndexer chainIndexDb eff
+         >> handleAssetClassFromAddressIndexer chainIndexDb eff
```
Comment (ak3n) [inline plutus-chain-index-core/src/Plutus/ChainIndex.hs:109]: Hmmm, does it mean that all handlers should be run in one transaction?

If the app will be stopped in the middle of handlers, half of data will be written but half will be lost and we will get a corrupted index.

Comment (ak3n) [reply plutus-chain-index-core/src/Plutus/ChainIndex.hs:109]: Yes, it helped to combine all sql effects to execute in one transaction.

**Comment by koslambrou** [comment]
Closing as not relevant anymore.

**Pull request closed without merging**